pylearn2/datasets/svhn.py

```diff
@@ -214,11 +233,11 @@ def split_train_valid(path, num_valid_train=400,
             train_index = list(train_index)
 
             train_x = numpy.concatenate((train_x,
-                                         data['X'][:, :, :, train_index]),
+                                        data['X'][:, :, :, train_index]),
                                         axis=3)
             train_y = numpy.concatenate((train_y, data['y'][train_index, :]))
             valid_x = numpy.concatenate((valid_x,
-                                         data['X'][:, :, :, valid_index]),
+                                        data['X'][:, :, :, valid_index]),
```

Comment:
I don't understand. If `y` consists in one-hot vectors, it would be in VectorSpace, but if it contains the indices, then it should be in IndexSpace.
Or are you saying that this conversion happens outside of the iterator?